### PR TITLE
Add purchase options card extensions

### PR DESCRIPTION
**Pull request by JulienHe**
### Background

Closes https://github.com/Shopify/subscriptions-app-remix/issues/1642, shipping for 2025-01, but will make it on unstable mid-october

Adding the UI extension action for the purchase options card in the product page and product variant page.
The extension target are named:

- `admin.product-purchase-option.action.render`  
- `admin.product-variant-purchase-option.action.render`  

This should be release as the same as we are releasing our new extensions by enabling our [beta flag](https://experiments.shopify.com/flags/subscriptions_admin_ui_extension).

We do also have a [PR for the documentation](https://github.com/Shopify/shopify-dev/pull/48780) of these targets.
[This PR](https://github.com/Shopify/shopify-dev/pull/49080) add the related documentations on Shopify-dev
 
### Checklist

- [x] I have :tophat:'d these changes
- [x] I have updated relevant documentation

### CODE REVIEW

File: packages/ui-extensions/src/surfaces/admin/api/purchase-options-card-action/purchase-options-card-action.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import type {StandardApi} from '../standard/standard';
+import type {ExtensionTarget as AnyExtensionTarget} from '../../extension-targets';
+
+export interface PurchaseOptionsCardConfigurationApi<
+  ExtensionTarget extends AnyExtensionTarget,
+> extends StandardApi<ExtensionTarget> {
```
Comment (MitchLillie) [inline packages/ui-extensions/src/surfaces/admin/api/purchase-options-card-action/purchase-options-card-action.ts:6]: I think you will want to extend `ActionExtensionApi` here, or just add the `close` method from that type (I forget if `extends` can override a property, in this case `data`.

**Comment by JulienHe** [comment]
> I think we have to wait to merge this until the `2024-10` branch has been created

Yes

File: packages/ui-extensions/src/surfaces/admin/api/purchase-options-card-action/purchase-options-card-action.ts (hunk 1 above)
Comment (JulienHe) [reply packages/ui-extensions/src/surfaces/admin/api/purchase-options-card-action/purchase-options-card-action.ts:6]: Correct, changed.

File: packages/ui-extensions/docs/surfaces/admin/screenshots/admin.product-purchase-option.action.render.png
[the remark this answers is not in the document]
Comment (JulienHe) [reply packages/ui-extensions/docs/surfaces/admin/screenshots/admin.product-purchase-option.action.render.png:1]: Sure, will change that 👍

**Review by vividviolet — APPROVED** [review]

**Review by paulomarg — APPROVED** [review]

**Review by MitchLillie — APPROVED** [review]

**Pull request merged**